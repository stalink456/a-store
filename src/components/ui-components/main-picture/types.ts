export type MainPictureType = {
  src: string;
  text: string;
  route: string;
};
