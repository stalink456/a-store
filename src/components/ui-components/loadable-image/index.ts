export { LoadableImage } from './loadable-image';
