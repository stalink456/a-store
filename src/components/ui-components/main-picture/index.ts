export { MainPicture } from "./main-picture";
