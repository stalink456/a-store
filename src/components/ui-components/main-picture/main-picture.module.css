@import "../../../css/variables.css";
@import "../../../css/animation.css";

.main__picture {
  position: relative;
  transition: all 0.5s;
  composes: opacity from global;
}

.main__picture img {
  width: 100%;
  height: 90vh;
  object-fit: cover;
  object-position: center 20%;
}

.main__picture h1 {
  position: absolute;
  top: 20px;
  left: 20px;
  color: var(--black-color);
  z-index: 1;
}

.main__picture:hover {
  filter: brightness(80%);
}

@media (max-width: 960px) {
  .main__picture h1 {
    top: 10px;
    left: 10px;
    font-size: 28px;
  }

  .main__picture img {
    height: 40vh;
  }
}
