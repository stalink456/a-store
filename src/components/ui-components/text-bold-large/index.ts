export { TextBoldLarge } from "./text-bold-large";
