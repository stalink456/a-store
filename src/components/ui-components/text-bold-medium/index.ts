export { TextBoldMedium } from "./text-bold-medium";
