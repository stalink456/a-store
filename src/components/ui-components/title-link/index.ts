export { TitleLink } from './title-link';
