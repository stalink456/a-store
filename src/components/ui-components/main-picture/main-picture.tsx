import React from "react";
import { Space } from "@alfalab/core-components/space";
import { TitleLink } from "components/ui-components/title-link";
import { LoadableImage } from "components/ui-components/loadable-image";
import { MainPictureType } from "./types";

import styles from "./main-picture.module.css";

export const MainPicture: React.FC<MainPictureType> = ({
  src,
  text,
  route,
}) => {
  return (
    <React.Fragment>
      <Space className={styles.main__picture} direction="horizontal" fullWidth>
        <TitleLink tag="h1" weight="bold" text={text} route={route}>
          <LoadableImage src={src} alt={text} />
        </TitleLink>
      </Space>
    </React.Fragment>
  );
};
