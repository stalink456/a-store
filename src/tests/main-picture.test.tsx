import React from 'react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MainPicture } from 'components/ui-components/main-picture';
import MadeInAlfa from 'pages/made-in-alfa';
import * as products from './mocks/get-products-200.json';
import * as loading from './mocks/get-loading-false.json';
import * as image from './mocks/get-image-200.json';

const mockProducts = products['products'];
const mockIsLoading = loading['loading'];
const mockedImage = image['image'];

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

jest.mock('hooks/use-made-in-alfa', () => ({
  useMadeInAlfa: () => ({
    products: mockProducts,
    isLoading: mockIsLoading,
  }),
}));

describe('MainPicture component', () => {
  it('should render MainPicture component', () => {
    const { baseElement } = render(
      <MainPicture
        src={mockedImage}
        text='Сделано в Альфе'
        route={'/made-in-alfa'}
      />,
      { wrapper: MemoryRouter }
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('should render MainPicture component with title', async () => {
    render(
      <MainPicture
        src={mockedImage}
        text='Сделано в Альфе'
        route={'/made-in-alfa'}
      />,
      { wrapper: MemoryRouter }
    );

    const result = await screen.findByText('Сделано в Альфе');

    expect(result).toBeInTheDocument();
  });

  it('should render MainPicture component with transition to MadeInAlfaPage', async () => {
    render(
      <MemoryRouter>
        <Routes>
          <Route
            path='/'
            element={
              <MainPicture
                src={mockedImage}
                text='Сделано в Альфе'
                route={'/made-in-alfa'}
              />
            }
          />
          <Route path='/made-in-alfa' element={<MadeInAlfa />} />
        </Routes>
      </MemoryRouter>
    );

    fireEvent.click(await screen.findByText('Сделано в Альфе'));

    expect(
      await screen.findByText('Футболка с бархатными стикерами')
    ).toBeInTheDocument();
  });

  it("shouldn't render MainPicture component with non-existent image", async () => {
    render(
      <MainPicture
        src={mockedImage}
        text='Сделано в Альфе'
        route={'/made-in-alfa'}
      />,
      { wrapper: MemoryRouter }
    );

    const image = screen.queryByAltText(
      'non-existent image'
    ) as HTMLImageElement;
    await waitFor(() => expect(image === null).toBeTruthy());
  });

  it("shouldn't render MainPicture component with non-existent description", async () => {
    render(
      <MainPicture
        src={mockedImage}
        text='Сделано в Альфе'
        route={'/made-in-alfa'}
      />,
      { wrapper: MemoryRouter }
    );

    const result = screen.queryByAltText(
      'non-existent description'
    ) as HTMLImageElement;
    await waitFor(() => expect(result).not.toBeTruthy());
  });
});
