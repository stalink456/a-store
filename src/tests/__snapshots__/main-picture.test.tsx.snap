// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPicture component should render MainPicture component 1`] = `
<body>
  <div>
    <div
      class="space__spaceContainer_nr3as space__start_nr3as space__spaceContainerFullWidth_nr3as main__picture"
    >
      <div
        class="space__spaceItem_nr3as space__spaceItemFullWidth_nr3as"
      >
        <a
          href="/made-in-alfa"
        >
          <h1
            class="typography__component_19zgr typography__styrene-medium_pxbut typography__bold_pxbut"
          >
            Сделано в Альфе
          </h1>
          <div
            class="skeleton__component_1bshy skeleton__animate_1bshy"
          >
            <img
              alt="Сделано в Альфе"
              class=""
              src="http://qa-games.ru/astore/public/images/61646585.png"
            />
          </div>
        </a>
      </div>
    </div>
  </div>
</body>
`;
